From: Thomas Gleixner <tglx@linutronix.de>
Date: Fri, 15 Jun 2018 18:28:05 +0200
Subject: cpu/hotplug: Split do_cpu_down()
Git-commit: 049ac01a7b9da9676b802fb13732da43cf6f25ca
Patch-mainline: v4.18 or v4.18-rc2 (next release)
References: bsc#1089343

Split out the inner workings of do_cpu_down() to allow reuse of that
function for the upcoming SMT disabling mechanism.

No functional change.

Signed-off-by: Thomas Gleixner <tglx@linutronix.de>
Cc: Borislav Petkov <bp@suse.de>
Cc: Ingo Molnar <mingo@kernel.org>
Cc: Lai Jiangshan <jiangshanlai@gmail.com>
Cc: Peter Zijlstra <peterz@infradead.org>
Cc: Thomas Gleixner <tglx@linutronix.de>
Cc: lkml <linux-kernel@vger.kernel.org>
Link: http://lkml.kernel.org/r/20180615162947.758649097@linutronix.de
Reviewed-by: Konrad Rzeszutek Wilk <konrad.wilk@oracle.com>
[ s/enum cpuhp_state/int/ - it is not used anyway. ]
Signed-off-by: Borislav Petkov <bp@suse.de>
---
 kernel/cpu.c |   18 +++++++++++-------
 1 file changed, 11 insertions(+), 7 deletions(-)

--- a/kernel/cpu.c
+++ b/kernel/cpu.c
@@ -281,20 +281,24 @@ out_release:
 	return err;
 }
 
+/*
+ * @target unused.
+ */
+static int cpu_down_maps_locked(unsigned int cpu, int target)
+{
+	if (cpu_hotplug_disabled)
+		return -EBUSY;
+	return _cpu_down(cpu, 0);
+}
+
 int __ref cpu_down(unsigned int cpu)
 {
 	int err;
 
 	cpu_maps_update_begin();
 
-	if (cpu_hotplug_disabled) {
-		err = -EBUSY;
-		goto out;
-	}
-
-	err = _cpu_down(cpu, 0);
+	err = cpu_down_maps_locked(cpu, 0);
 
-out:
 	cpu_maps_update_done();
 	return err;
 }
